<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_6">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>6</number>
      </property>
      <property name="sizeConstraint">
       <enum>QLayout::SetMinAndMaxSize</enum>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="7,3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QGroupBox" name="groupBox_dns_or_ip">
            <property name="title">
             <string>Host地址</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QTextEdit" name="textEdit_dns_or_ip">
               <property name="acceptDrops">
                <bool>false</bool>
               </property>
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;从界面中生成host和port清单的函数，支持多种格式&lt;/p&gt;&lt;p&gt;行内用空格分开如：www.baidu.com 192.168.1.1&lt;/p&gt;&lt;p&gt;对C类ip地址支持范围地址如：192.168.1.1-254&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="autoFormatting">
                <set>QTextEdit::AutoNone</set>
               </property>
               <property name="tabChangesFocus">
                <bool>true</bool>
               </property>
               <property name="lineWrapMode">
                <enum>QTextEdit::WidgetWidth</enum>
               </property>
               <property name="html">
                <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;127.0.0.1&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QGroupBox" name="groupBox_port">
            <property name="title">
             <string>端口</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QTextEdit" name="textEdit_port">
               <property name="acceptDrops">
                <bool>false</bool>
               </property>
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;port同理，同行用空格分开，支持范围表达式如：1-65535&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="tabChangesFocus">
                <bool>true</bool>
               </property>
               <property name="html">
                <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;20-25 53 69 80 89 443 110 111 &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;137 139 445 143 161 389 512-514&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;1443 3306 1521 2049 3389 3690&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;5000 5432 5900-5902 6379&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;7001-7002 9200 9300 &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;27017-27018 11211 50070 50030&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;8080-9100&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="7,3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QGroupBox" name="groupBox_result">
            <property name="title">
             <string>扫描结果</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <item>
              <widget class="QCheckBox" name="checkBox_show_only_valid">
               <property name="focusPolicy">
                <enum>Qt::ClickFocus</enum>
               </property>
               <property name="text">
                <string>仅显示有效结果结果/显示所有结果</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QTextEdit" name="textEdit_result">
               <property name="focusPolicy">
                <enum>Qt::ClickFocus</enum>
               </property>
               <property name="acceptDrops">
                <bool>false</bool>
               </property>
               <property name="tabChangesFocus">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_config">
          <property name="title">
           <string>配置</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_15">
           <item>
            <widget class="QTabWidget" name="tabWidget_config">
             <property name="tabletTracking">
              <bool>false</bool>
             </property>
             <property name="focusPolicy">
              <enum>Qt::ClickFocus</enum>
             </property>
             <property name="tabShape">
              <enum>QTabWidget::Rounded</enum>
             </property>
             <property name="currentIndex">
              <number>0</number>
             </property>
             <property name="elideMode">
              <enum>Qt::ElideLeft</enum>
             </property>
             <widget class="QWidget" name="tab_6">
              <attribute name="title">
               <string>并发</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_14">
               <item>
                <widget class="QGroupBox" name="groupBox_6">
                 <property name="title">
                  <string>线程数量配置</string>
                 </property>
                 <layout class="QVBoxLayout" name="verticalLayout_15">
                  <item>
                   <layout class="QVBoxLayout" name="verticalLayout_16">
                    <item>
                     <widget class="QRadioButton" name="radioButton_single_thread">
                      <property name="focusPolicy">
                       <enum>Qt::ClickFocus</enum>
                      </property>
                      <property name="text">
                       <string>单线程</string>
                      </property>
                      <property name="checked">
                       <bool>false</bool>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <layout class="QHBoxLayout" name="horizontalLayout_17">
                      <item>
                       <widget class="QRadioButton" name="radioButton_muti_hreads">
                        <property name="focusPolicy">
                         <enum>Qt::ClickFocus</enum>
                        </property>
                        <property name="text">
                         <string>多线程并发量</string>
                        </property>
                        <property name="checked">
                         <bool>true</bool>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_threads_amount">
                        <property name="enabled">
                         <bool>true</bool>
                        </property>
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="focusPolicy">
                         <enum>Qt::ClickFocus</enum>
                        </property>
                        <property name="text">
                         <string>128</string>
                        </property>
                        <property name="alignment">
                         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tab_5">
              <attribute name="title">
               <string>速率</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_11">
               <item>
                <widget class="QGroupBox" name="groupBox_5">
                 <property name="title">
                  <string>TCP连接超时</string>
                 </property>
                 <layout class="QVBoxLayout" name="verticalLayout_12">
                  <item>
                   <layout class="QVBoxLayout" name="verticalLayout_13">
                    <item>
                     <widget class="QRadioButton" name="radioButton_default_delay">
                      <property name="focusPolicy">
                       <enum>Qt::ClickFocus</enum>
                      </property>
                      <property name="text">
                       <string>默认超时(500ms)</string>
                      </property>
                      <property name="checked">
                       <bool>true</bool>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <layout class="QHBoxLayout" name="horizontalLayout_16">
                      <item>
                       <widget class="QRadioButton" name="radioButton_customize_delay">
                        <property name="focusPolicy">
                         <enum>Qt::ClickFocus</enum>
                        </property>
                        <property name="text">
                         <string>超时时间</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="lineEdit_delay">
                        <property name="enabled">
                         <bool>true</bool>
                        </property>
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="focusPolicy">
                         <enum>Qt::ClickFocus</enum>
                        </property>
                        <property name="text">
                         <string>50</string>
                        </property>
                        <property name="alignment">
                         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLabel" name="label">
                        <property name="text">
                         <string>ms</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QProgressBar" name="progressBar">
          <property name="value">
           <number>0</number>
          </property>
          <property name="invertedAppearance">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_progress">
          <property name="text">
           <string>0/0</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_total">
          <property name="text">
           <string>共计</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_interrupt">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="text">
           <string>强制退出</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_reset">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="text">
           <string>清空信息</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_scan">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="text">
           <string>扫描(F5)</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>扫描</string>
    </property>
    <addaction name="action_scan"/>
    <addaction name="action_Abort"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_scan">
   <property name="text">
    <string>开始</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="action_clear">
   <property name="text">
    <string>清空 </string>
   </property>
   <property name="shortcut">
    <string>F6</string>
   </property>
  </action>
  <action name="action_Abort">
   <property name="text">
    <string>强制退出</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="action_new">
   <property name="text">
    <string>新建</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_save">
   <property name="text">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_saveAs">
   <property name="text">
    <string>另存为</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_Exit">
   <property name="text">
    <string>退出</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>textEdit_dns_or_ip</tabstop>
  <tabstop>textEdit_port</tabstop>
  <tabstop>textEdit_result</tabstop>
  <tabstop>checkBox_show_only_valid</tabstop>
  <tabstop>tabWidget_config</tabstop>
  <tabstop>radioButton_default_delay</tabstop>
  <tabstop>radioButton_customize_delay</tabstop>
  <tabstop>lineEdit_delay</tabstop>
  <tabstop>radioButton_single_thread</tabstop>
  <tabstop>radioButton_muti_hreads</tabstop>
  <tabstop>lineEdit_threads_amount</tabstop>
  <tabstop>pushButton_interrupt</tabstop>
  <tabstop>pushButton_reset</tabstop>
  <tabstop>pushButton_scan</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_reset</sender>
   <signal>clicked()</signal>
   <receiver>textEdit_dns_or_ip</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>658</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>220</x>
     <y>201</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_reset</sender>
   <signal>clicked()</signal>
   <receiver>textEdit_port</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>680</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>661</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_reset</sender>
   <signal>clicked()</signal>
   <receiver>textEdit_result</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>687</x>
     <y>550</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>432</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
